setup.py

```diff
@@ -53,6 +53,7 @@ def create_version_file():
     'fastparquet==0.4.1',
     'scikit-optimize',
     'catboost>=0.23.0,<0.25',
+    'xgboost==1.2',
```

Comment:
do other versions of xgboost cause issue?

Reply:
Yes. With custom evaluation metric function, other versions transform predictions in an unintended way ([link](https://github.com/dmlc/xgboost/pull/5954)) and may cause errors on our codes. And XGBoost team plan to change some their API on coming 1.3.0. I fix the version to 1.2 which is a current stable version.